crypto/keyring/legacy_info.go

```diff
@@ -1,50 +1,51 @@
 package keyring
```

Comment:
how about doing like in module, and creating a `crypto/keyring/migrations` package. Then all migration logic (including this legacyInfo) can go in that package.

Reply:
what do you think, @robert-zaremba ?

Reply:
> how about doing like in module, and creating a `crypto/keyring/migrations` package. Then all migration logic (including this legacyInfo) can go in that package.

What do you wanna to put inside `crypto/keyring/migrations` package? `record.go`, `record_test.go` and `legacy_info.go` ? It is worth noting that `Migrate` and `MigrateAll()` methods are located inside `keyring` package.

Reply:
```
func (ks keystore) MigrateAll() (bool, error) {
    return migrations.MigrateAll(ks.db, ks.cdc)
}
```

I started moving legacy_info.go and migration logic (`MigrateAll` and `Migrate` funcs from keyring.go) into `crypto/keyring/migrations` package. if i dont move `record.go` into `migrations` package, it introduces `import cycle dep  issue` since I require access to `Record` struct inside `migrations` package . May I move 'record.go` into `migrations` package as well? It would be great to move `types.pb.go` and `types.go` to `migrations` package as well.

Reply:
I like the idea of `migrations` package. But I think it will be challenging without breaking `keyring` package in sub-packages.
@cyberbono3  - migrations package shouldn't contain core types. It should only contain migration logic and legacy types.

So, if it will be too much of efforts, then I think it's not really worth.

Reply:
It is challenging due to cycle dep issue described above. From my perspective, it is not worth it in this case.

Reply:
Sounds good